### ISSUE TITLE
unary vector negation doesn't work

### ISSUE THREAD

**Issue by Dunbaratu** [KSP-KOS/KOS#341]
This:
``` set oppositeVector to - theVector. ```
doesn't work, but it should.  It should mean  the same thing as:
``` set oppositeVector to (-1) * theVector. ```

**Comment by Dunbaratu** [comment]
While I'm at it, I'm adding the "negate" operator to direction.  A negative Direction is its quaternion inverse.  (i.e. "rotate +20 degrees around x and +10 degrees around y" will become "rotate -20 degrees around x and -10 degrees around y").

I had also worked out a cool algorithm for getting a color negative from an RGB, that was reversible, as a negation operator should be, but then I balked at not knowing whether or not I wanted to have it also invert the alpha (transparent colors become opaque and visa-versa).  That would be *mathematically* correct, but very un-intuitive and weird.  (Basically the negative of the RGB value is simply "The color on the opposite side of the color cube from me, just as far from the color cube's center (rgb(0.5,0.5,0.5)) as I am."  That's basically drawing a "vector" to the color, with the color cube's center as the origin, and negating that vector.  Thus black becomes white, red becomes cyan, yellow becomes blue, and saturation is preserved (slightly tinted grey becomes slightly tinted in the opposing color).  Mathematically it's just subtracting each r,g,b component from 1.  (i.e. a red value of 0.25 becomes a red value of 0.75.)  In "HSV" terms it would be a bit like leaving the S term as-is, and making H into 1-H and making V into 1-V, although you might have wanted to give HSV's a different negation algorithm.

Anyway, in the end I decided to shelve the "negative color" idea because I didn't know what to do about the alpha channel.
